### ISSUE TITLE
Moving many mails fails due to "too long argument"

### ISSUE THREAD

**Issue by tribut** [roundcube/roundcubemail#6875]
When trying to move the result of a search query (>10000 mails), the move fails with the error message

    Server error: UID COPY: Error in IMAP command UID COPY: Too long argument (0.001 + 0.043 + 0.042 secs).

I don't think this is something I can fix on the server side, so I guess it would have to move the messages in batches?

Roundcube 1.3.9, server is Dovecot 2.2.27

**Comment by alecpl** [comment]
Could you provide imap_debug log for this operation. So, I can see what limit exactly is exceeded. We should in general use short form of uid list, but maybe it's not used in this scenario or indeed the list cannot be made short enough for a single IMAP command.

**Comment by alecpl** [comment]
You could also check `imap_max_line_length` in dovecot config.

**Comment by tribut** [comment]
Sure. I eventually used "doveadm move" to split the messages yesterday, but I managed to reproduce the behavior using a similar folder:

~~~
[31-Jul-2019 13:58:39 +0200]: <804f9838> [440F] S: A0003 OK [READ-WRITE] Select completed (0.000 + 0.000 secs).
[31-Jul-2019 13:58:39 +0200]: <804f9838> [440F] C: A0004 UID COPY 439,698:704,706:709,715:718,881:883,1030,3049,3052,3064:3065,3072,3388,3394,3399:3409,3431:3432,4405,4644:4645,4651,4660:4661,4664,4668,4694,4749,6
217,6219,7132,7177,15066,15069,15075:15076,15078:15082,15088,15105,15110,15113:15115,15149,15173,15249,15251:15253,15291,15306,15310:15311,15315,15334,15341,15344,15348:15349,15354:15355,15358:15359,15361,15366:15
367,15370:15372,15381,15383:15384,15389,15396:15397,15400:15401,15403,15405,15408,15421,15429:15430,15434,15436,15454,15464,15480,15496,15515,15532:15533,15605,15607:15611,15613,15616,15619,15624:15625,15637,15643
,15646,15648,15656,15665,15745,15769,15785,15790:15791,15793,15797:15799,15801,15807,15809,15812,15818,15824,15841,15843:15844,15849,15851,15854:15858,15861,15863,15867:15873,15875,15878:15880,15882:15883,15885,15
890:15892,15894,15906,15910,15926:15927,15932,15936,15944:15945,15954,15959:15960,15962,15965,15970,15980,15986,15988:15990,15992,15994:15995,15997,16000,16002,16007,16017,16067:16071,16120,16128,16133,16140:16141
,16157,16187,16198,16208,16229,16244:16245,16249,16252,16254:16255,16258,16278:16280,16283,16289,16296:16297,16299,16306,16317,16322,16345,16358,16360,16365:16366,16368,16370:16371,16373,16393,16417,16425,16432,16
436,16471,16475:16476,16486:16487,16490,16498,16500:16501,16505:16508,16510,16512,16518:16519,16521:16522,16526,16528:16529,16535,16539,16543,16546,16551,16565,16572,16584,16598,16610,16738,16746:16747,16817,16821
,16837,16846,16859,16881,16921,16924,16926,16930,16955:16956,16962,16969,16982,16991,16995,16997,17077,17132,17137,17144,17153,17157,17161,17164,17202:17203,17234,17265:17266,17271:17273,17281:17282,17284,17294,17
299,17305,17307:17308,17312,17328,17336,17381,17559:17560,17562,17606,17609,17611,17642:17644,17646,17648:17649,17651:17652,17655,17657:17658,17661,17741,17752,17755:17756,17759,17762,17773,17775,17822,17830,17842
,17844,17923,17943,17964,18034,18040:18041,18051,18061,18064:18066,18073,18077,18086,18096,18103,18109,18111:18119,18125,18128:18130,18133:18135,18137:18140,18142:18144,18146:18148,18150,18155:18158,18161,18173:18
174,18176,18187:18192,18198,18233,18312:18313,18315,18317,18325,18328,18335,18346:18348,18357,18367,18376,18380,18387,18401,18427:18428,18435:18436,18443,18466,18468,18509,18532:18533,18543,18568,18577,18579,18584
:18586,18591:18593,18612,18629,18633,18635,18646,18671:18672,18707,18742,18745,18748:18749,18752:18755,18759,18761,18765:18766,18768:18769,18772:18774,18776:18777,18784:18787,18801,18804,18808,18826,18838:18839,18
858,18869,18872:18873,18878,18889,18892,18925,18927,18933,18937:18938,18940,18989,18999,19006:19007,19011:19012,19018,19021,19034:19036,19042,19045,19056,19072,19075,19080,19088,19119,19127,19139:19141,19145:19146
,19148:19149,19152:19154,19164:19165,19171,19180:19181,19188,19193,19197:19198,19210:19211,19213:19214,19217,19228,19232,19234,19236:19240,19245:19246,19248,19250,19253,19255,19263:19264,19274,19279,19292:19294,19
299,19301:19304,19312:19314,19317,19319,19322,19324,19327:19329,19333,19339,19342:19343,19345,19347:19349,19351,19363,19376:19378,19382:19383,19388,19397,19399:19400,19403:19405,19408,19413,19415,19417,19429,19437
,19441,19449,19461,19467,19506:19507,19514,19518,19524,19543,19546:19547,19550,19554,19578:19584,19586,19589,19591,19595:19598,19603,19608,19614,19618,19622,19624,19627,19634,19639,19642,19649,19671:19672,19676,19
685,19692,19705:19706,19709,19713:19714,19719,19728:19729,19741,19743:19744,19759,19761,19764,19772,19776,19778,19793,19806,19821,19824,19836,19839,19868,19878,19880,19885,19892,19907,19913,19915,19917,19919,19929
,19932:19933,19940,19947,19950,19971:19972,19984,19992,20000:20002,20038:20039,20042:20046,20049:20050,20055,20070,20084:20086,20088,20098,20117,20126,20144,20147,20156,20159,20168,20170:20171,20174,20196:20197,20
205:20206,20238,20241,20252,20254,20278,20287,20310,20322,20520,20524,20557,20563,20567,20589,20615,20619,20649,20652:20653,20658,20663,20671,20678,20686,20770,20777,20791,20935:20939,21021:21022,21027,21031,21066
:21067,21081:21082,21096:21097,21118,21122,21131,21140:21141,21151,21186,21206,21308,21317,21323,21... [truncated 79366 bytes]
[31-Jul-2019 13:58:39 +0200]: <804f9838> [440F] S: A0004 BAD Error in IMAP command UID COPY: Too long argument (0.001 + 0.001 secs).
[31-Jul-2019 13:58:39 +0200]: <804f9838> [440F] C: A0005 LOGOUT
~~~

Increasing `imap_max_line_length` does indeed solve the problem, so I guess other than the error message being somewhat unhelpful, this is probably not a bug.

Thanks!
